src/etl/impl/SubscriptionSource.cpp

```diff
@@ -240,7 +240,11 @@ SubscriptionSource::handleMessage(std::string const& message)
 
         } else {
             if (isForwarding_) {
-                if (object.contains(JS(transaction))) {
+                // Clio as rippled's proposed_transactions subscirber, will receive two jsons for each transaction
+                // 1 - Proposed transaction
+                // 2 - Validated transaction
+                // Only forward proposed transaction, validated transactions are sent by Clio itself
+                if (object.contains(JS(transaction)) and !object.contains(JS(meta))) {
```

Comment:
Is `!object.contains(JS(meta))` the best way to check whether the transaction is proposed? Doesn't it have some explicit field for that?

Reply:
Unfortunately , there is not. I have noticed the difference by observation. https://github.com/XRPLF/rippled/blob/d5e5c3c2206ea7790ec5fb140355345e46f6755f/src/ripple/app/misc/NetworkOPs.cpp#L2759C9-L2759C18